rfc5424/syslog_message.go

```diff
@@ -88,15 +88,27 @@ func (sm *SyslogMessage) Valid() bool {
 	return false
 }
 
-// SetPriority set the priority value and the computed facility and severity codes accordingly.
-func (sm *SyslogMessage) SetPriority(value uint8) {
+func (sm *SyslogMessage) setPriority(value uint8) {
 	sm.Priority = &value
 	facility := uint8(value / 8)
 	severity := uint8(value % 8)
 	sm.facility = &facility
 	sm.severity = &severity
 }
 
+// SetPriority set the priority value and the computed facility and severity codes accordingly.
+//
+// It enforces a correct priority value (range [0, 191]), since it can not by null by RFC.
+func (sm *SyslogMessage) SetPriority(value uint8) *SyslogMessage {
+	if value >= 0 && value <= 191 {
+		sm.setPriority(value)
+
+		return sm
+	}
+
+	panic("out of range priority value")
```

Comment:
I don't think this should panic.

I think you would mark the message as invalid or return a nil or something.

Reply:
Chosen to ignore out of range values simply not setting them.

This setters always return a `SyslogMessage` to have a fluent style.